include/deal.II/grid/tria_accessor.templates.h

```diff
@@ -3030,19 +3036,20 @@ CellAccessor<dim,spacedim>::is_locally_owned () const
 {
   Assert (this->active(),
           ExcMessage("is_locally_owned() can only be called on active cells!"));
-#ifndef DEAL_II_WITH_P4EST
+#ifndef DEAL_II_WITH_MPI
   return true;
 #else
   if (is_artificial())
     return false;
 
-  const parallel::distributed::Triangulation<dim,spacedim> *pdt
-    = dynamic_cast<const parallel::distributed::Triangulation<dim,spacedim> *>(this->tria);
+  const parallel::Triangulation<dim,spacedim> *pt
+    = dynamic_cast<const parallel::Triangulation<dim,spacedim> *>(this->tria);
 
-  if (pdt == 0)
+  if (pt == 0)
     return true;
   else
-    return (this->subdomain_id() == pdt->locally_owned_subdomain());
+    return (this->subdomain_id() == pt->locally_owned_subdomain());
```

Comment:
Yes, this approach is clearly beautiful. No need to even distinguish between the two parallel triangulations :-)